src/main/java/com/treasuredata/client/model/TDJob.java

```diff
@@ -108,6 +108,53 @@ public boolean isFinished()
         }
     }
 
+    /**
+     * Engine version for hive, presto
+     * To accept specific version number like 'Hive TD 2018.01', Implement as not enum but class
+     */
+    public static class EngineVersion
+    {
+        private final String engineVersion;
+
+        private EngineVersion(String engineVersion)
+        {
+            this.engineVersion = engineVersion;
+        }
+
+        public static final EngineVersion EXPERIMENTAL = new EngineVersion("experimental");
+        public static final EngineVersion STABLE = new EngineVersion("stable");
+        public static final EngineVersion LEGACY = new EngineVersion("legacy");
+        public static final EngineVersion CURRENT = new EngineVersion("current");
+
+        @JsonValue
+        public String getEngineVersion()
+        {
+            return engineVersion;
+        }
+
+        @Override
+        public String toString()
+        {
+            return engineVersion;
+        }
+
+        public static EngineVersion fromString(String engineVersion)
+        {
+            switch (engineVersion) {
+                case "stable":
+                    return STABLE;
+                case "experimental":
+                    return EXPERIMENTAL;
+                case "legacy":
+                    return LEGACY;
+                case "current":
+                    return CURRENT;
+                default:
```

Comment:
There are cases were other versions might show up in the future, so this behavior could be confusing. While I like the idea of having static defaults available, it would probably be a good idea to allow any string as the version.

Reply:
Thanks for your comment. 
EngineVersion.fromString() accepts any string such as "TD Hive 2019.1" and others and return EngineVersion instance.
EXPERIMENTAL, STABLE and others are only provided for user's convenience.
If I make misunderstand, please let me know.

Reply:
Thanks for your comment. 
EngineVersion.fromString() accepts any string such as "TD Hive 2019.1" and others and return EngineVersion instance.
EXPERIMENTAL, STABLE and others are only provided for user's convenience.
If I make misunderstand, please let me know.

Reply:
Ah, sorry it's already doing that... my brain isn't working correctly today 😕